### PR TITLE
fix lint warning

### PR DESCRIPTION
**Pull request by jillxuu**
```
jillxu@Jills-MacBook-Pro explorer % pnpm lint

> aptos-explorer@0.0.1 lint /Users/jillxu/Documents/GitHub/explorer
> tsc --noEmit && eslint src --ext ts,tsx --report-unused-disable-directives --max-warnings 0
```

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,247 +1,173 @@
-lockfileVersion: '6.0'
+lockfileVersion: 5.4
```
Comment (geekflyer) [inline pnpm-lock.yaml:1]: please update your pnpm version to 8.x and revert this lockfile change

File: src/api/hooks/useGetSearchResults.ts [hunk 2]
```diff
@@ -143,15 +140,9 @@ export default function useGetSearchResults(input: string) {
       }
 
       const resultsList = await Promise.all(promises);
-      const results = resultsList
-        .filter((result): result is SearchResult => !!result)
-        .map((result) => ({
-          ...result,
-          to:
-            result.to !== null
-              ? augmentToWithGlobalSearchParams(result.to)
```
Comment (geekflyer) [inline src/api/hooks/useGetSearchResults.ts:152]: you sure we can remove this?

File: src/global-config/network-selection.ts [hunk 3]
```diff
@@ -42,35 +42,31 @@ export function useNetworkSelector() {
...
+        (prev) => {
+          const newParams = new URLSearchParams(prev);
+          newParams.set("network", network);
+          return newParams;
+        },
+        {replace},
+      );
+      writeSelectedNetworkToLocalStorage(network);
     },
-    [], // empty [] makes this effect only run once (on mount)
```
Comment (geekflyer) [inline src/global-config/network-selection.ts:71]: I think this is a case where we most likely want the previous behaviour and only run this effect on initial mount. Perhaps add some `// eslint-ignore <rule-name>` here to stop the linter from complaining about it.

Let's just undo the code change in this file and add this eslint ignore comment

File: src/pages/Account/Tabs/ModulesTab/Tabs.tsx [hunk 4]
```diff
@@ -93,7 +93,7 @@ function ModulesTabs({address}: {address: string}): JSX.Element {
         break;
     }
     eventName && logEvent(eventName);
-  }, [value]);
+  }, [value, logEvent]);
```
Comment (geekflyer) [inline src/pages/Account/Tabs/ModulesTab/Tabs.tsx:96]: looking at the implementation of useLogEvent it looks like it returns a new function on every rerender and hence would retrigger this effect every time, which is probably not what we want.

File: src/pages/Validators/Index.tsx [hunk 5]
```diff
@@ -54,7 +53,15 @@ export default function ValidatorsPage() {
         },
       );
     }
-  }, []);
```
Comment (geekflyer) [inline src/pages/Validators/Index.tsx:57]: might be another case where we want to trigger the event only once / on initial mount? not sure though without digging deeper into the logic

**Review by geekflyer — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml (hunk 1 above)
Comment (jillxuu) [reply pnpm-lock.yaml]: pnpm version is alraedy 8.6.0 but if i run `pnpm i` this file will be generated again with version 5.4

Comment (geekflyer) [reply pnpm-lock.yaml]: are you sure? that sounds very odd

Comment (jillxuu) [reply pnpm-lock.yaml]: https://github.com/aptos-labs/explorer/assets/121921928/175cd260-9a72-4402-a7e4-adddffa5ab8e

Comment (geekflyer) [reply pnpm-lock.yaml]: you forgot the most important part in the video - `pnpm --version` :-P

**Review by 0xZihan — APPROVED** [review]

**Pull request merged**